q2cli/builtin/tools.py

```diff
@@ -195,7 +195,89 @@ def peek(path):
         click.echo(metadata.format)
 
 
-@tools.command('inspect-metadata',
+@tools.command(name='cast-metadata',
+               short_help='Designate metadata columns as numerical'
+                          ' or categorical.',
+               help='Designating the column types within a metadata file.'
+                    ' Supported column types are numerical or categorical.'
+                    ' Providing multiple file paths to this command will merge'
+                    ' the metadata.',
+               cls=ToolCommand)
+# NOTE: # This gives a potential way to avoid multiple --cast calls but it's
+# fiddely https://stackoverflow.com/a/47730333
+@click.option('--cast', required=True, multiple=True,
+              help='Cast flags for each metadata column that should be'
+              ' specified as either categorical or numeric column types.')
+@click.option('--error-on-extra', is_flag=True,
+              help='Cast flags provided inline that do not exist within the'
+              ' metadata provided will result in a raised error.'
+              ' Enabled by default. If this flag is disabled and'
```

Comment:
> Enabled by default

I think there might be an error somewhere in here, because there is no obvious way (to me at least) to _disable_ this parameter:

```bash
(q2dev)
~/projects/qiime2/q2cli
☻  cat md1.tsv                                                                                                 cast_metadata_tool 6ccc7c7 ✗
id      foo     bar
a       1       one
b       2       two
c       3       three
(q2dev)
~/projects/qiime2/q2cli
☻  qiime tools cast-metadata md1.tsv --cast foo:categorical spleen:numeric --output-file md1out.tsv            cast_metadata_tool 6ccc7c7 ✗
Usage: qiime tools cast-metadata [OPTIONS] METADATA...
...
                    There was a problem with the command:
 (1/1) Invalid value for 'METADATA...': File 'spleen:numeric' does not exist.
(q2dev)
~/projects/qiime2/q2cli
☻  qiime tools cast-metadata md1.tsv --cast foo:categorical spleen:numeric --output-file md1out.tsv --error-on-extra
Usage: qiime tools cast-metadata [OPTIONS] METADATA...
...
                    There was a problem with the command:
 (1/1) Invalid value for 'METADATA...': File 'spleen:numeric' does not exist.
(q2dev)
~/projects/qiime2/q2cli
☻  qiime tools cast-metadata md1.tsv --cast foo:categorical spleen:numeric --output-file md1out.tsv --error-on-extra=False
Usage: qiime tools cast-metadata [OPTIONS] METADATA...
...
                    There was a problem with the command:
 (1/1?) --error-on-extra option does not take a value
```

Reply:
@thermokarst thanks for pointing this out! These flags do need to be modified a bit - will get this updated as well.

Reply:
I would like to see this problem resolved without the introduction of two additional flags to this command - I am concerned that that will make the user-facing API a bit more complicated than it needs to be here. Please take a look at the `--show-importable-types` and `--show-importable-formats` flags on the `qiime tools import` command for some inspiration on solving this using only two flags.

Reply:
Also, take a look at the docs here for a nice example: https://click.palletsprojects.com/en/8.0.x/options/#boolean-flags

Reply:
@thermokarst sounds good - we've updated the logic for these flags, and reverted back to just using two flags in total.